packages/commerce-sdk-react/src/hooks/query.test.tsx

```diff
@@ -0,0 +1,211 @@
+    //     hook: () => useCustomerPaymentInstrument({customerId: '123', paymentInstrumentId: '456'}),
+    //     endpoint: '/customers/123/payment-instruments/456',
+    // },
+    // {
+    //     name: 'useCustomerProductLists',
+    //     hook: () => useCustomerProductLists({customerId: '123'}),
+    //     endpoint: '/customers/123/product-lists',
+    // },
+    // {
+    //     name: 'useCustomerProductList',
+    //     hook: () => useCustomerProductList({customerId: '123'}),
+    //     endpoint: '/customers/123/product-lists/456',
+    // },
+    // {
+    //     name: 'useCustomerProductListItem',
+    //     hook: () => useCustomerProductListItem({customerId: '123'}),
+    //     endpoint: '/customers/123/product-lists/456/items/789',
+    // },
+    // {
+    //     name: 'usePublicProductListsBySearchTerm',
+    //     hook: () => usePublicProductListsBySearchTerm(),
+    //     endpoint: '',
+    // },
+    // {
+    //     name: 'usePublicProductList',
+    //     hook: () => usePublicProductList(),
+    //     endpoint: '',
+    // },
+    // {
+    //     name: 'useProductListItem',
+    //     hook: () => useProductListItem(),
+    //     endpoint: '',
+    // },
+
+
+    {
+        name: 'useOrder',
+        hook: () => useOrder({orderNo: '123'}),
+        endpoint: '/orders/123',
+    },
+    {
+        name: 'usePaymentMethodsForOrder',
+        hook: () => usePaymentMethodsForOrder({orderNo: '123'}),
+        endpoint: '/orders/123/payment-methods',
+    },
+    {
+        name: 'useTaxesFromOrder',
+        hook: () => useTaxesFromOrder({orderNo: '123'}),
+        endpoint: '/orders/123/taxes',
+    },
+
+
+    {
+        name: 'useProducts',
+        hook: () => useProducts({ids: '123,456'}),
+        endpoint: '/products',
+    },
+    {
+        name: 'useProduct',
+        hook: () => useProduct({id: '123'}),
+        endpoint: '/products/123',
+    },
+    {
+        name: 'useCategories',
+        hook: () => useCategories({ids: '123,456'}),
+        endpoint: '/categories',
+    },
+    {
+        name: 'useCategory',
+        hook: () => useCategory({id: '123'}),
+        endpoint: '/categories/123',
+    },
+    {
+        name: 'usePromotions',
+        hook: () => usePromotions({ids: '123,456'}),
+        endpoint: '/promotions',
+    },
+    {
+        name: 'usePromotionsForCampaign',
+        hook: () => usePromotionsForCampaign({campaignId: '123'}),
+        endpoint: '/promotions/campaigns/123',
+    },
+
+    {
+        name: 'useProductSearch',
+        hook: () => useProductSearch({q: 'test'}),
+        endpoint: '/product-search',
+    },
+    {
+        name: 'useSearchSuggestions',
+        hook: () => useSearchSuggestions({q: 'test'}),
+        endpoint: '/search-suggestions',
+    },
+]
+
+test.each(QUERY_TESTS)('%j - 200 returns data', async ({hook, endpoint}) => {
+    const data = {test: true}
+    nock('http://localhost:8888')
+        .get((uri) => uri.includes(endpoint))
```

Comment:
`includes` is not precise enough. It may accidentally matches `uri` that is unexpected.

Reply:
well, since we call `nock.cleanAll` after each tests, and for every query unit tests there is only one http call, so i'm not too worried about this... Any suggestions for a more precise version?

Reply:
I think we should be worried about it. It can lead to a bug that is not immediately visible. 

It's possible that during copy and paste, we forgot to update the endpoint (or the hook). So for `uri.includes('/orders/123')`, the expression will be true for `uri` like '/orders/123/payment-methods',  '/orders/123/taxes'.

Reply:
We can use regular expression like `/orders\/123$/` to be more precise

Reply:
Good call @vmarta , agreed on using regex `$` to avoid the conflict 🏅

Reply:
thanks for the advice, changes are made.